src/textureProcessing/TextureProximityProcessor.h

```diff
@@ -0,0 +1,93 @@
+/** Copyright (C) 2016 Tim Kuipers - Released under terms of the AGPLv3 License */
+#ifndef TEXTURE_PROCESSING_TEXTURE_PROXIMITY_PROCESSOR_H
+#define TEXTURE_PROCESSING_TEXTURE_PROXIMITY_PROCESSOR_H
+
+#include <vector>
+
+#include "../utils/intpoint.h"
+#include "../utils/SparseLineGrid.h"
+
+#include "../settings/settings.h"
+
+#include "../slicer/SlicerSegment.h"
+#include "TexturedMesh.h"
+
+namespace cura
+{
+
+/*!
+ * Class for recording texture coordinates at places where textures are defined, for later looking in the proximity of a texture.
+ */
+class TextureProximityProcessor
+{
+public:
+    /*!
+     * Helper class to retrieve and store texture to bump map settings
+     */
+    struct Settings
+    {
+        coord_t proximity; //!< The distance within which to search for nearby texture
+        Settings(coord_t proximity)
+        : proximity(proximity)
+        {
+        }
+    };
```

Comment:
These are algorithm parameters, not user settings.

This is a construction to avoid function calls with really long parameter lists.
Instead of passing all kinds of parameters directly to the function with all kinds of default values, we pass an object to the algorithm which can have different constructors.
That way we can also avoid function calls of `Alg::f(Obj& base, Obj& model, bool param1 = false, bool param2 = false, bool param3 = false)` with `Alg::f(base, model, false, false, true)`. Instead we can do `Alg::f(base, model, Alg::getParam3Type())` or something.

However, this is all very premature if the only algorithm parameter turns out to be just a simple type.

I do think it's a good architecture design, though.
What do you think?

Reply:
I agree that it's good architecture. Not blindly, but it makes sense to use this as a way to group parameters together. Give the struct a good name. It doesn't really make sense if it's a group of 1 parameter though. Just `coord_t` would make sense in this case, unless this is meant to be expanded with new settings soon.

Reply:
Okay. I have disolved the type.